lambdas/build-user-identity/src/main/java/uk/gov/di/ipv/core/builduseridentity/BuildUserIdentityHandler.java

```diff
@@ -148,6 +139,13 @@ public APIGatewayProxyResponseEvent handleRequest(
                     clientOAuthSessionItem.getGovukSigninJourneyId());
 
             String userId = clientOAuthSessionItem.getUserId();
+
+            var scopeClaims = clientOAuthSessionItem.getScope().split(" ");
+            if (!input.getPath().contains(USER_IDENTITY_ENDPOINT)
+                    || !Arrays.asList(scopeClaims).contains(OPENID)) {
+                return ApiGatewayResponseGenerator.getAccessDeniedApiGatewayProxyResponseEvent();
+            }
```

Comment:
This will add some extra validation on the scope in prod, so out of an abundance of paranoia can we put this behind the MFA_RESET feature flag?

(You were away, but we had a P1 incident last week from the validation on `scope` we added at the start of the journey 😬 )